### PR TITLE
cmake: remove usage of zephyr_library_ifdef 

### LINKED ISSUE #7565: zephyr_library_ifdef has unexpected behaviour
zephyr_library_ifdef is defined like this:

https://github.com/zephyrproject-rtos/zephyr/commit/c674167fea42ba75e795cc4d2dcdbf410f1a2ea7

it is unexpected behaviour for it to modify the callers control-flow (return() out of the CMakeLists.txt file).

### PR DESCRIPTION
**Pull request by AdithyaBaglody**
All instances of the cmake macro zephyr_library_ifdef will be removed in this PR.

[SEBO] EDIT: Fixes #7565

### CODE REVIEW

**Comment by SebastianBoe** [comment]
Why DNM?

**Comment by SebastianBoe** [comment]
The definition of "zephyr_library_ifdef" should also be removed in this PR.

**Comment by AdithyaBaglody** [comment]
I am adding a few more commits to this PR. 
I want to check how the CI behaves before I push all of them. 
There is also an issue #7663 which I  want to check before pushing anything.

**Comment by galak** [comment]
Why are we removing it, that doesn't seem to be explained anywhere.

**Comment by nashif** [comment]
> Why are we removing it, that doesn't seem to be explained anywhere.

#7565

**Comment by SebastianBoe** [comment]
> Why are we removing it, that doesn't seem to be explained anywhere.

Because that would resolve

https://github.com/zephyrproject-rtos/zephyr/issues/7565

**Comment by AdithyaBaglody** [comment]
@galak the commit which explain the reason for removing the cmake macro hasn't been pushed yet. will be sent in the subsequent patches.

File: boards/arm/cc3220sf_launchxl/cc3220sf_launchxl_defconfig [hunk 1]
```diff
@@ -4,7 +4,6 @@ CONFIG_SOC_FAMILY_TISIMPLELINK=y
 CONFIG_SOC_SERIES_CC32XX=y
 CONFIG_SOC_CC3220SF=y
 CONFIG_CORTEX_M_SYSTICK=y
-CONFIG_FLASH=y
```
Comment (GAnthony) [inline boards/arm/cc3220sf_launchxl/cc3220sf_launchxl_defconfig]: I validated this for cc3220sf.  Looks good.

Comment (AdithyaBaglody) [reply boards/arm/cc3220sf_launchxl/cc3220sf_launchxl_defconfig]: Thanks for the confirmation.

**Comment by AdithyaBaglody** [comment]
Removing the DNM since this fixes #7565 
For the issue #7663 I will create a new PR when I get the solution.

File: drivers/CMakeLists.txt [hunk 2]
```diff
@@ -23,16 +22,17 @@ add_subdirectory_if_kconfig(pinmux)
...
 add_subdirectory_if_kconfig(can)
 
+add_subdirectory_ifdef(CONFIG_FLASH_HAS_DRIVER_ENABLED flash)
+add_subdirectory_ifdef(CONFIG_SERIAL_HAS_DRIVER serial)
 add_subdirectory_ifdef(CONFIG_AIO_COMPARATOR   aio)
 add_subdirectory_ifdef(CONFIG_BT               bluetooth)
 add_subdirectory_ifdef(CONFIG_NETWORKING       net)
 add_subdirectory_ifdef(CONFIG_NET_L2_ETHERNET  ethernet)
-add_subdirectory_ifdef(CONFIG_ENTROPY_GENERATOR entropy)
+add_subdirectory_ifdef(CONFIG_ENTROPY_HAS_DRIVER entropy)
```
Comment (SebastianBoe) [inline drivers/CMakeLists.txt]: Will changing these semantics cause entropy drivers, that used to be pruned, to now be included into applications?

Is there any way of proving yes / no. E.g. by running sanitycheck and comparing sizes before and after.

The reason I suspect there is a risk of this is that this build script used to say:

"Include the entropy driver when the entropy generator is enabled."

But now it says:

"Include the entropy driver when the entropy generator is supported."

**Review by SebastianBoe — CHANGES_REQUESTED** [review]

File: drivers/CMakeLists.txt (hunk 2 above)
Comment (AdithyaBaglody) [reply drivers/CMakeLists.txt]: @SebastianBoe I understand the issue here. But this Kconfig "entropy has driver" will only be enabled if entropy_generator is enabled. So we will maintain the same subdirectory include condition as before.

Comment (SebastianBoe) [reply drivers/CMakeLists.txt]: I see, then this is safe.

**Review by SebastianBoe — APPROVED** [review]

**Pull request merged**